src/test/java/seedu/module/logic/commands/TagCommandTest.java

```diff
@@ -41,10 +43,38 @@ void execute_addTagUnfilteredList_success() {
         for (int i = 0; i < firstTask.getTags().size(); i++) {
             expectedTags[i] = firstTask.getTags().iterator().next().tagName;
         }
-        expectedTags[firstTask.getTags().size()] = TAG_STUB;
+        expectedTags[expectedTags.length - 1] = TAG_STUB;
         Task editedTask = new TaskBuilder(firstTask).withTags(expectedTags).build();
 
-        TagCommand tagCommand = new TagCommand(INDEX_FIRST_TASK, new Tag(TAG_STUB));
+        TagCommand tagCommand = new TagCommand(INDEX_FIRST_TASK);
+        Set<Tag> tagStubs = new HashSet<>();
+        tagStubs.add(new Tag(TAG_STUB));
+        tagCommand.setTags(tagStubs);
+
+        String expectedMessage = String.format(TagCommand.MESSAGE_TAG_TASK_SUCCESS, editedTask);
+
+        Model expectedModel = new ModelManager(new ModuleBook(model.getModuleBook()), new UserPrefs());
+        expectedModel.setTask(firstTask, editedTask);
+
+        assertCommandSuccess(tagCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    void execute_addMultipleTagUnfilteredList_success() {
+        Task firstTask = model.getFilteredTaskList().get(INDEX_FIRST_TASK.getZeroBased());
+        String[] expectedTags = new String[firstTask.getTags().size() + 2];
```

Comment:
Perhaps it'd be better to use variables and methods to decide the size of String[] expectedTags, since "multiple" doesn't always mean 2.

Reply:
I have changed this to addTwoTags.... In this case, two and multiple have no difference, and adds a set of methods just for this test is a bit.. wasteful?